[CoreRenderingService] Add dev docs (#218630)
elastic/kibana: dev_docs/shared_ux/shared_ux_landing.mdx

      title: 'Kibana Page Template',
      description: "Learn how to use Kibana Page Template for your app's page layout",
    },
    {
      pageId: 'kibDevRenderingService',
      title: 'Core Rendering Service',

>       description: 'Learn how to provide the global context needed for rendering custom React elements',

Reviewer: Maybe?
```suggestion
      description: 'Learn how to provide the Kibana global context in your React apps',
```
